python_modules/dagster/dagster/_daemon/auto_run_reexecution/auto_run_reexecution.py

```diff
@@ -23,56 +24,19 @@
 
 
 def filter_runs_to_should_retry(
-    runs: Sequence[DagsterRun], instance: DagsterInstance, default_max_retries: int
+    runs: Sequence[DagsterRun], instance: DagsterInstance
 ) -> Iterator[Tuple[DagsterRun, int]]:
     """Return only runs that should retry along with their retry number (1st retry, 2nd, etc.)."""
-
-    def get_retry_number(run: DagsterRun) -> Optional[int]:
-        if run.status != DagsterRunStatus.FAILURE:
-            return None
-
-        raw_max_retries_tag = run.tags.get(MAX_RETRIES_TAG)
-        if raw_max_retries_tag is None:
-            max_retries = default_max_retries
-        else:
-            try:
-                max_retries = int(raw_max_retries_tag)
-            except ValueError:
-                instance.report_engine_event(
-                    f"Error parsing int from tag {MAX_RETRIES_TAG}, won't retry the run.", run
-                )
-                return None
-
-        if max_retries == 0:
-            return None
-
-        # TODO: group these to reduce db calls
-        run_group = instance.get_run_group(run.run_id)
-
-        if run_group:
-            _, run_group_iter = run_group
-            run_group_list = list(run_group_iter)
-
-            # Has the parent run already been retried the maximum number of times? (Group includes the parent)
-            if len(run_group_list) >= max_retries + 1:
-                return None
-
-            # Does this run already have a child run?
-            if any([run.run_id == run_.parent_run_id for run_ in run_group_list]):
-                return None
-            return len(run_group_list)
-        else:
-            return 1
-
-    default_retry_on_asset_or_op_failure: bool = instance.run_retries_retry_on_asset_or_op_failure
-
     for run in runs:
-        retry_number = get_retry_number(run)
-        retry_on_asset_or_op_failure = get_boolean_tag_value(
-            run.tags.get(RETRY_ON_ASSET_OR_OP_FAILURE_TAG),
-            default_value=default_retry_on_asset_or_op_failure,
-        )
-        if retry_number is not None:
+        if get_boolean_tag_value(run.tags.get(WILL_RETRY_TAG), default_value=False):
+            if not get_boolean_tag_value(run.tags.get(DID_RETRY_TAG), default_value=False):
+                _, run_group = check.not_none(instance.get_run_group(run.run_id))
+                yield (run, len(list(run_group)))
+        else:
+            retry_on_asset_or_op_failure = get_boolean_tag_value(
```

Comment:
this else statement needs to stay in sync with the corresponding part of `should_retry_run` so that we can log the engine event if necessary. thinking about way to refactor this so that we dont rely on remembering to keep the conditions the same